### PR TITLE
Metrics master

### PR DESCRIPTION
**Pull request by guljain**
This PR does 2 things:

- Currently the instrumentation contained metrics that could be per GHFS instance based but now they are also at executor level. This means that there is one metric per executor also. 

Adding GCS specific statistics :

- gcs_client_side_error_count : Counts the occurrence of client side error status code
- gcs_server_side_error_count: Counts the occurrence of server side error status code
- gcs_client_rate_limit_count: Counts the occurrence of 429 status code
- gcs_total_request_count: Counts the total number of gcs requests made
- exception_count: Counts the number of exceptions encountered (Can be expanded in future)

### CODE REVIEW

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsGlobalStorageStatistics.java [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+import org.apache.hadoop.util.functional.CallableRaisingIOE;
+
+/** Storage statistics for GCS */
+@InterfaceAudience.Private
+@InterfaceStability.Unstable
+public class GhfsGlobalStorageStatistics extends StorageStatistics {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  /** {@value} The key that stores all the registered metrics */
+  public static final String NAME = "GhfsGlobalStorageStatistics";
```
[the remark this answers is not in the document]
Comment (guljain) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsGlobalStorageStatistics.java:76]: There is a pre-existing GhfsStorageStatistics.java so I changed this. Can change the FS based GhfsStorageStatistics to GhfsFsStorageStatistics.java then. Does that work ?

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopOutputStreamTest.java [hunk 2]
```diff
@@ -53,14 +54,19 @@ public class GoogleHadoopOutputStreamTest {
 
   private GoogleHadoopFileSystem ghfs;
 
+  // private static GoogleCloudStorageFileSystemIntegrationHelper gcsFsIHelper;
+
   @Before
   public void setUp() throws IOException {
     ghfs = GoogleHadoopFileSystemTestHelper.createInMemoryGoogleHadoopFileSystem();
+    // gcsFsIHelper = GoogleCloudStorageFileSystemIntegrationHelper.create();
```
[the remark this answers is not in the document]
Comment (guljain) [reply gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopOutputStreamTest.java:62]: Because we are not really using gcsFsIHelper in the tests.

**Review by arunkumarchacko — APPROVED** [review]

**Pull request merged**